mu-plugins/blocks/table-of-contents/src/view.js

```diff
@@ -65,26 +65,58 @@ function onScroll() {
 	}
 }
 
-function init() {
-	const container = document.querySelector( '.wp-block-wporg-table-of-contents' );
+function isSidebarWithinViewport( container ) {
 	// Margin offset from the top of the sidebar.
 	const gap = getCustomPropValue( '--wp--preset--spacing--edge-space' );
+	// Usable viewport height.
+	const viewHeight = window.innerHeight - FIXED_HEADER_HEIGHT;
+	// Get the height of the sidebar, plus the top margin and 50px for the
+	// "Back to top" link, which isn't visible until `is-fixed-sidebar` is
+	// added, therefore not included in the parentNode.offsetHeight value.
+	const sidebarHeight = container.parentNode?.offsetHeight + gap + 50;
+	// If the table of contents sidebar is shorter than the view area, apply the
+	// class so that it's fixed and scrolls with the page content.
+	return sidebarHeight < viewHeight;
+}
+
+function init() {
+	const container = document.querySelector( '.wp-block-wporg-table-of-contents' );
 
 	if ( container ) {
-		// Usable viewport height.
-		const viewHeight = window.innerHeight - FIXED_HEADER_HEIGHT;
-		// Get the height of the sidebar, plus the top margin and 50px for the
-		// "Back to top" link, which isn't visible until `is-fixed-sidebar` is
-		// added, therefore not included in the parentNode.offsetHeight value.
-		const sidebarHeight = container.parentNode?.offsetHeight + gap + 50;
-		// If the table of contents sidebar is shorter than the view area, apply the
-		// class so that it's fixed and scrolls with the page content.
-		if ( sidebarHeight < viewHeight ) {
+		if ( isSidebarWithinViewport( container ) ) {
 			container.parentNode.classList.add( 'is-fixed-sidebar' );
 			onScroll(); // Run once to avoid footer collisions on load (ex, when linked to #reply-title).
 			window.addEventListener( 'scroll', onScroll );
 		}
 	}
+
+	const toggleButton = container?.querySelector( '.wporg-table-of-contents__toggle' );
+	const list = container?.querySelector( '.wporg-table-of-contents__list' );
+
+	if ( toggleButton && list ) {
+		toggleButton.addEventListener( 'click', function () {
+			if ( toggleButton.getAttribute( 'aria-expanded' ) === 'true' ) {
+				toggleButton.setAttribute( 'aria-expanded', false );
+				list.setAttribute( 'style', 'display:none;' );
+			} else {
+				toggleButton.setAttribute( 'aria-expanded', true );
+				list.removeAttribute( 'style' );
+			}
+
+			// After toggle, see if we need to update the sidebar classes.
+			if ( isSidebarWithinViewport( container ) ) {
+				container.parentNode.classList.add( 'is-fixed-sidebar' );
+			} else {
+				container.parentNode.classList.remove( 'is-fixed-sidebar' );
+				window.scrollTo( { top: 0, left: 0, behavior: 'instant' } );
+			}
+			container.parentNode.classList.remove( 'is-bottom-sidebar' );
+			const isBottom = onScroll();
```

Comment:
It's not exactly clear what (or why) this would return a value. Can we add a documentation block to the `onScroll` function since I think this is the first time the return value is assigned?

Reply:
Added more docs in https://github.com/WordPress/wporg-mu-plugins/pull/389/commits/3f6ed32fba62899956e857c2ec86fac1e43c9a0e